src/software/ai/navigator/obstacle/new_obstacle/circle_obstacle.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "software/ai/navigator/obstacle/new_obstacle/obstacle.h"
+#include "software/new_geom/circle.h"
+
+/**
+ * An Obstacle represented by a Circle
+ */
+class CircleObstacle : public Obstacle
+{
+   public:
+    CircleObstacle() = delete;
+
+    /**
+     * Construct a circle obstacle by with a Circle
+     *
+     * @param circle Circle to make obstacle with
+     */
+    explicit CircleObstacle(const Circle circle);
+
+    bool contains(const Point& p) const override;
+    double minDistance(const Point& p) const override;
+    bool intersects(const Segment& segment) const override;
+    std::string toString(void) const override;
+
+    /**
+     * Gets the underlying circle
+     *
+     * @return circle
+     */
+    const Circle getCircle(void) const;
+
+   private:
+    Circle circle_;
```

Comment:
Perhaps we could have `CircleObstacle` extend both `Obstacle` and `Circle` so that we don't need to manually wrap functions?

Reply:
looked into it and it's difficult to support inheritance->can't think of a good way to write getCircle(). Generally, I think composition is a better design pattern than inheritance

Reply:
https://en.wikipedia.org/wiki/Composition_over_inheritance

Reply:
Sorry, just to be clear here, what would the difficulty of supporting inheritance be here? At a quick glance it seems like we would be able to just make it extend `Circle` and then delete the extra wrapper functions we have?

Reply:
Extending `Circle` would only give us `contains` and we would need to construct a representation of the superclass (can't think of a "nice" way to do this) to implement distance and intersects, since they use the geom util functions. The benefit here is one function at the cost of a lot more complexity.

Reply:
Sorry, I think I'm misunderstanding something here, so just bear with me. I changed `CircleObstacle` to extend `public Circle`, and then wrote (and ran) the following test:
```
TEST(NavigatorObstacleTest, circle_distance){
    Circle expected({2, 2}, 3);
    CircleObstacle circle_obstacle(expected);

    ::distance(circle_obstacle, Point(0,0));
}
```
and it worked fine. The same thing should apply to the other functions we've wrapped here. 

I'm happy to call about this if you think it would be easier/faster to discuss that way!

Reply:
Ah I totally misunderstood what you meant. I didn't realize that you wanted to change the interface so that the caller directly calls util functions. That makes a lot more sense

Reply:
@garethellis0 I remember why I designed it this way now. The goal is to have contains, distance, and intersects to work on `Obstacle`s without needing to know what type (Circle/Polygon) the Obstacle is. We would need a function like ::distance(Obstacle o, Point p), which is difficult to implement without knowing the underlying shape of obstacle, hence the inheritance structure we have here

Reply:
Ok, that's reasonable, so I'm good with this as-is.